src/raftstore/store/peer.rs

```diff
@@ -415,6 +423,35 @@ impl Peer {
         Ok(())
     }
 
+    pub fn is_request_allowed(&self, req: &RaftCmdRequest) -> bool {
+        // Read and write is allowed when the peer is leader, readonly request is allowed
+        // although when peer is follower.
+        if !self.is_leader() &&
+           (!self.is_readonly_query(req) || self.leader_id() == INVALID_ID || self.is_applying()) {
+            return false;
+        }
+
+        true
+    }
+
+    pub fn is_readonly_query(&self, req: &RaftCmdRequest) -> bool {
+        if req.get_requests().len() == 0 {
+            return false;
+        }
+
+        for cmd_req in req.get_requests() {
+            if cmd_req.get_cmd_type() != CmdType::Snap && cmd_req.get_cmd_type() != CmdType::Get {
+                return false;
+            }
+        }
+
+        if self.get_store().applied_index_term != self.raft_group.raft.term {
```

Comment:
my fault, there is no need check `applied_index_term` and `raft.term` neither for leader nor for follower, because `ReadIndexResp` return committed index.